pkg/k8s/resource/resource.go

```diff
@@ -354,6 +362,11 @@ func (r *resource[T]) startWhenNeeded() {
 		return
 	}
 
+	// Wait for CRDs to have synced before trying to access (Cilium) k8s resources
+	if r.opts.crdSyncPromise != nil {
+		r.opts.crdSyncPromise.Await(r.ctx)
```

Comment:
Nit. Should return in case of an error, as that means that the context expired and the component is shutting down.

Reply:
will have to leave for later.